app/controllers/admin/customers_controller.rb

```diff
@@ -17,9 +17,10 @@ def index
       respond_to do |format|
         format.html
         format.json do
-          render_as_json @collection,
-                         tag_rule_mapping: tag_rule_mapping,
-                         customer_tags: customer_tags_by_id
+          render json: @collection,
+                 each_serializer: ::Api::Admin::CustomerWithBalanceSerializer,
```

Comment:
The serializer's class name is to me a clear indication that we should extract an API endpoint out of this, together with the if branch of #collection.